src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -673,3 +676,328 @@ The Run handle/object supports few other use-cases.
+           xrtGraphReset(graphHandle);
+           
+           // run the graph infinitely
+           xrtGraphRun(graphHandle, -1);
+           
+           xrtGraphWait(graphHandle,3);  // Suspends the graph after 3 AIE cycles from the previous start 
+           
+           
+           xrtGraphResume(graphHandle); // Restart the suspended graph again to run forever
+           
+           xrtGraphSuspend(graphHandle); // Suspend the graph immediately
+           
+           xrtGraphResume(graphHandle); // Restart the suspended graph again to run forever
+           
+           xrtGraphEnd(graphHandle,5);  // End the graph operation after 5 AIE cycles from the previous start
+
+
+In the example above
+
+- The API ``xrtGraphRun(graphHandle, -1)`` is used to execute the graph infinitely
+- The API ``xrtGraphWait`` suspends the graph after 3 AIE cycles from the graph starts. 
+
+       - If the graph was already run more than 3 AIE cycles the graph is suspended immediately. 
+- The API ``xrtGraphResume`` is used to restart the suspended graph
+- The API ``xrtGraphSuspend`` is used to suspend the graph immediately
+- The API ``xrtGraphEnd(graphHandle,5)`` is  ending the graph after 5 AIE cycles from the previous graph start. 
+       
+       - If the graph was already run more than 5 AIE cycles the graph is suspended immediately.
+       - Using ``xrtGraphEnd`` eliminates the capability of rerunning the Graph (without loading PDI and a graph reset again). 
+
+
+Measuring AIE cycle consumed by the Graph
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The API ``xrtGraphTimeStamp`` can be used to determine AIE cycle consumed between a graph start and stop. 
+
+Here in this example, the AIE cycle consumed by 3 iteration is calculated
+ 
+
+.. code:: c++
+      :number-lines: 35
+           
+           // start from reset state
+           xrtGraphReset(graphHandle);
+           
+           uint64_t begin_t = xrtGraphTimeStamp(graphHandle);
+           
+           // run the graph for 3 iteration
+           xrtGraphRun(graphHandle, 3);
+           
+           xrtGraphWait(graphHandle, 0); 
+           
+           uint64_t end_t = xrtGraphTimeStamp(graphHandle);
+           
+           std::cout<<"Number of AIE cycles consumed in the 3 iteration is: "<< end_t-begin_t; 
+           
+
+RTP (Runtime Parameter) control
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+XRT provides the API to update and read the runtime parameters of the graph. 
+
+- The API ``xrtGraphUpdateRTP`` to update the RTP 
+- The API ``xrtGraphReadRTP`` to read the RTP. 
+
+.. code:: c++
+      :number-lines: 35
+
+           ret = xrtGraphReset(graphHandle);
+           if (ret) throw std::runtime_error("Unable to reset graph");
+
+           ret = xrtGraphRun(graphHandle, 2);
+           if (ret) throw std::runtime_error("Unable to run graph");
+
+           float increment[1] = {1};
+           const char *inVect = reinterpret_cast<const char *>(increment);
```

Comment:
Ouch! Perhaps an API not requiring this line?
Hint: use a templated function or generic lambda to implement `xrtGraphUpdateRTP`.